.github/workflows/backend/deploy-preview-backend.yml

```
@@ -0,0 +1,54 @@
+name: Deploy Preview
+on:
+  workflow_run:
+    workflows: ["Test Backend"]
+    types:
+      - completed
+    branches: ['main']
+    paths:
+      - 'backend/**'
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: backend
+    steps:
+      - name: Deploy to Preview Server
+        uses: appleboy/ssh-action@v1.0.3
+        env:
+          MICONFIG: ${{ secrets.MICONFIG_PREVIEW }}
+          PK: ${{ secrets.BENTO_PK_PREVIEW }}
+        with:
+          host: ${{ secrets.PREVIEW_SERVER_IP }}
+          username: ${{ secrets.PREVIEW_USER }}
+          key: ${{ secrets.PREVIEW_SERVER_SSH_KEY }}
+          envs: MICONFIG,PK
+          script: |
+            clean_up() {
+              echo "Performing cleanup..."
+              cd ~
+              echo "Remove repository"
+              rm -rf repo
+              echo "Docker cleanup"
+              docker system prune -f
+              echo "Done: Cleanup"
+            }
+            handle_error() {
+              clean_up
+              echo "Failed to deploy preview"
+              exit 1
+            }
+            trap 'handler_error' ERR
```

WOOOOOOOOOO

Everything is great, the trap for error handling just has a typo

```suggestion
            trap 'handle_error' ERR
```